content/root/best-practices-guide.textile

```diff
@@ -0,0 +1,97 @@
+
+h4. 5. Using API key on the client side
+
+* Ably's "Basic Authentication":/general/authentication#basic-authentication method is the easiest way to authenticate your client on the Ably Realtime platform. However, it directly exposes the API key in the source code. When you use it on the client side, any entity that interacts with the client could view your API key and potentially misuse it. If you can run an auth server, we recommend using "Token Authentication":/general/authentication#token-authentication on the client side. "Read more":/general/authentication#selecting-auth about selecting an authentication mechanism.
+
+h4. 6. Ensuring tokens are auto&#45;renewable
+
+* Tokens are short lived and need to be constantly renewed. Rather than getting a token from your auth servers yourself and initializing the Client Library SDK with it, we recommend you instead initialize the SDK with an authUrl or authCallback. It will use this information to automatically request a token from your auth server's endpoint whenever it needs a new one, thus seamlessly switching to the new token without disconnecting, hence ensuring an uninterrupted service.
+
+h4. 7. Choosing between Realtime/ REST libraries and Protocol Adapters 
+
+* Want to maintain a persistent connection with Ably's Realtime platform, so you can subscribe to messages sent on a channel, or be notified whenever the presence set changes? Use the "realtime library":/realtime.
+* Want to just publish data to the platform, don't care about subscribing, and don't want to manage channel attachment? Use the "REST library":/rest.
+* Want to publish at very high volumes or with the lowest possible latency? "The realtime libary can publish at higher volumes (due to message pipelining) and lower latencies than is possible with the REST library":https://support.ably.io/solution/articles/3000060319-should-i-use-the-rest-or-realtime-library.
+* Want to get one&#45;time info about the presence set of a channel? Use the "REST library":/rest.
+* Want to set up realtime communication with IoT devices on platforms for which there's no native Ably client library available? Use the "MQTT protocol adapter":/mqtt.
+* Want to migrate from PubNub/Pusher? Use the relevant "protocol adapter":/adapters to "ease the pain of migration":https://support.ably.io/solution/articles/3000054127-can-you-help-ease-the-pain-of-migration-from-another-realtime-service: you can switch to Ably right away while sticking with your current client libraries, and gradually migrate to native Ably clients.
```

Comment:
You've not mentioned subscribe to events on a server.  I realise it's mentioned below, but is this not about chosing the right library?